src/log.ts

```diff
@@ -25,6 +25,25 @@ const LOG_ALL_STACKTRACES =
 const LOG_FORMAT = env.varOrDefault('LOG_FORMAT', 'simple');
 const INSTANCE_ID = env.varOrUndefined('INSTANCE_ID');
 
+type ItemFilterChildLoggerOpts = {
+  itemFilter: string;
+};
+
+// in the filters.ts file, we want to build a tree of filters
+// so we concatinate previous filter string to the next one
+const appendItemFilterFormat = format((info, opts: unknown) => {
+  if (
+    typeof opts === 'object' &&
+    typeof (opts as ItemFilterChildLoggerOpts).itemFilter === 'string'
+  ) {
+    const optsFromChildLogger = opts as ItemFilterChildLoggerOpts;
+    info.itemFilter = info.itemFilter
+      ? `${info.itemFilter} AND ${optsFromChildLogger.itemFilter}`
```

Comment:
I was trying to figure out exactly what this does and looking for `AND` in the output and didn't see it. Do you have an example log message form you testing where this shows up? It's quite possible it's there and I'm just not seeing it.

Reply:
first: this is silly code because if I pass itemFilter, it will have overridden and itemFilter previously in the logger context.
Secondly, I've been going back and forth on this. And we don't need this I think. All the "or" and "and" higherarchy is already captured in the map operators in createFilter and we pass logger to each one. So we are ok here.
I'm adding deeply nested "and" "or "tree" and I don't see any loss of contex. OTOH this isn't very plesantly to debug with the json strings compressed into 1 liners. But that's different topic.

So, I'm removing this logger filter for now.